File: models/Gpt-3.5-turbo/cpp/code/problems/range_queries/Subarray_Sum_Queries.cpp

#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;

    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    int curr_sum = 0, max_sum = 0;
    for (int i = 0; i < n; i++) {
        curr_sum += arr[i];
        max_sum = max(max_sum, curr_sum);
        curr_sum = max(curr_sum, 0);
    }

    for (int i = 0; i < m; i++) {
        int k, x;
        cin >> k >> x;
        k--;

        curr_sum += x - arr[k];
        max_sum = max(max_sum, curr_sum);
        arr[k] = x;

        cout << max_sum << endl;
    }

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/range_queries/Subarray_Sum_Queries.py

n, m = map(int, input().split())
arr = list(map(int, input().split()))

max_sum = 0
current_sum = 0

for i in range(n):
    current_sum = max(arr[i], current_sum + arr[i])
    max_sum = max(max_sum, current_sum)
    print(max_sum)

for _ in range(m):
    k, x = map(int, input().split())
    k -= 1
    arr[k] = x

    if current_sum > 0:
        current_sum += x - arr[k]
    else:
        current_sum = max(x, current_sum + x)

    max_sum = max(max_sum, current_sum)
    print(max_sum)
